fl4health/clients/basic_client.py

```diff
@@ -387,14 +388,32 @@ def _handle_reporting(
         reporting_dict.update(metric_dict)
         self.wandb_reporter.report_metrics(reporting_dict)
 
-    def train_step(self, input: torch.Tensor, target: torch.Tensor) -> Tuple[TrainingLosses, Dict[str, torch.Tensor]]:
+    def _move_data_to_device(self, data: TorchInputType) -> TorchInputType:
+        if isinstance(data, torch.Tensor):
+            return data.to(self.device)
+        elif isinstance(data, dict):
```

Comment:
I believe we can be a little more specific with the type of dict here. That is, `Dict[str, torch.Tensor]`

Reply:
`isinstance(data, Dict[str, torch.Tensor])` actually doesn't work because `isinstance` can't take in Generic types. I know using `dict` is not ideal but I don't know of a better way...

Reply:
Right...I forgot about that part. Using dict here is sufficient then